main.go

```diff
@@ -143,11 +148,72 @@ func watchGGST(noClose bool, ctx context.Context) {
 	}
 }
 
+func downloadFile(filepath string, url string) error {
+	resp, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	out, err := os.Create(filepath)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	_, err = io.Copy(out, resp.Body)
+	return err
+}
+
+func selfDelete() {
+	var sI syscall.StartupInfo
+	var pI syscall.ProcessInformation
+	exePath, err := os.Executable()
+	if err != nil {
+		panic(err)
+	}
+	argv, err := syscall.UTF16PtrFromString(os.Getenv("windir") + "\\system32\\cmd.exe /C del " + exePath)
+	if err != nil {
+		panic(err)
+	}
+	err = syscall.CreateProcess(nil, argv, nil, nil, true, 0, nil, nil, &sI, &pI)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func autoUpdate() {
+	resp, err := http.Get(GithubReleasesLink)
+	if err != nil {
+		panic(err)
+	}
+
+	url := resp.Request.URL.String()
+	pattern := regexp.MustCompile(`v\d+.\d+.\d+`)
+	matches := pattern.FindStringSubmatch(url)
+
+	if len(matches) < 1 {
+		return
+	}
+
+	version := matches[0]
+	if version != Version {
+		fmt.Println("New version released, downloading")
+		err := downloadFile("totsugeki_"+version+".exe", GithubDownloadLink)
```

Comment:
Two problems with this:

1. It downloads the file in the current working directory, which may not be the directory Totsugeki is in.
2. If someone is using the unga-bunga version, it will download vanilla Totsugeki regardless of which version they're using.